File: src/main/java/LDS/Person/service/WikiCommentService.java
package LDS.Person.service;

import LDS.Person.dto.request.WikiCommentByWikiIdRequest;
import LDS.Person.dto.response.WikiCommentResponse;

import java.util.List;

/**
 * Wiki留言业务逻辑接口
 */
public interface WikiCommentService {

  /**
   * 根据wiki_id查询所有留言
   * 返回的留言包含对应用户的name和cover字段
   *
   * @param wikiId Wiki ID
   * @return Wiki留言响应列表
   */
  List<WikiCommentResponse> getCommentsByWikiId(Long wikiId);

  /**
   * 根据wiki_id查询所有留言（通过Request对象）
   *
   * @param request 查询请求
   * @return Wiki留言响应列表
   */
  List<WikiCommentResponse> getCommentsByWikiId(WikiCommentByWikiIdRequest request);

  /**
   * 删除Wiki留言记录
   *
   * @param replyId 留言ID
   * @return true 删除成功，false 删除失败（记录不存在）
   */
  boolean deleteComment(Long replyId);

  /**
   * 点赞留言
   *
   * @param replyId 留言ID
   * @return true 点赞成功，false 点赞失败
   */
  boolean likeComment(Long replyId);
}

File: src/Services/WikiCommentService.cs
using SidusX.DTOs.Request;
using SidusX.DTOs.Response;
using SidusX.Models;
using SidusX.Repositories;

namespace SidusX.Services;

public class WikiCommentService
{
    private readonly WikiCommentRepository _repo;
    private readonly UserRepository _userRepo;
    public WikiCommentService(WikiCommentRepository repo, UserRepository userRepo) { _repo = repo; _userRepo = userRepo; }

    public async Task<WikiCommentResponse> CreateAsync(WikiCommentCreateRequest req)
    {
        if (string.IsNullOrWhiteSpace(req.Text)) throw new ArgumentException("留言内容不能为空");
        var c = new WikiComment { WikiId = req.WikiId, UserId = req.UserId, Text = req.Text, CreateTime = DateTime.Now };
        c.ReplyId = await _repo.InsertAsync(c);
        return new WikiCommentResponse { ReplyId = c.ReplyId, WikiId = c.WikiId, UserId = c.UserId, Text = c.Text, CreateTime = c.CreateTime };
    }

    public async Task<bool> DeleteAsync(long replyId) => await _repo.DeleteAsync(replyId);

    public async Task LikeAsync(long replyId) => await _repo.IncrementLikesAsync(replyId);

    public async Task<PageResponse<WikiCommentResponse>> PageByWikiIdAsync(WikiCommentByWikiIdRequest req)
    {
        var (items, total) = await _repo.PageByWikiIdAsync(req.WikiId, req.Page, req.PageSize);
        var responses = new List<WikiCommentResponse>();
        foreach (var item in items)
        {
            var name = await _userRepo.GetNameByIdAsync(item.UserId);
            responses.Add(new WikiCommentResponse { ReplyId = item.ReplyId, WikiId = item.WikiId, UserId = item.UserId, UserName = name, Text = item.Text, Likes = item.Likes, CreateTime = item.CreateTime });
        }
        return new PageResponse<WikiCommentResponse> { Total = total, Page = req.Page, PageSize = req.PageSize, Data = responses };
    }
}
